recipes/brigand/all/conanfile.py

```diff
@@ -13,29 +14,51 @@ class BrigandConan(ConanFile):
     topics = ("meta-programming", "boost", "runtime", "header-only")
     homepage = "https://github.com/edouarda/brigand"
     license = "BSL-1.0"
-    settings = "compiler"
-    requires = "boost/1.75.0"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_boost": [True, False],
+    }
+    default_options = {
+        "with_boost": True,
+    }
+
     no_copy_sources = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def requirements(self):
+        if self.options.with_boost:
+            # TODO: add transitive_headers=True & bump required_conan_version to >=1.52.0
+            self.requires("boost/1.79.0")
+
+    def package_id(self):
+        self.info.clear()
 
     def validate(self):
-        if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, 11)
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 11)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def build(self):
+        pass
 
     def package(self):
-        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
         include_path = os.path.join("include", "brigand")
-        self.copy("*.hpp", dst=include_path, src=os.path.join(self._source_subfolder, include_path))
-
-    def package_id(self):
-        self.info.header_only()
+        copy(self, "*.hpp", src=os.path.join(self.source_folder, include_path), dst=os.path.join(self.package_folder, include_path))
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "libbrigand"
+        self.cpp_info.set_property("pkg_config_name", "libbrigand")
+        self.cpp_info.bindirs = []
+        self.cpp_info.frameworkdirs = []
+        self.cpp_info.libdirs = []
+        self.cpp_info.resdirs = []
+        if self.options.with_boost:
+            self.cpp_info.requires = ["boost::headers"]
+        else:
+            self.cpp_info.defines("BRIGAND_NO_BOOST_SUPPORT")
```

Comment:
Is this syntax correct? All other conanfiles say `self.cpp_info.defines = ["BRIGAND_NO_BOOST_SUPPORT"]`

Reply:
good catch